paddle/fluid/operators/cinn_launch_op.cc

```diff
@@ -62,90 +62,105 @@ void DebugCinnCompiledResult(const CinnCompiledObject& result) {
+      [](const auto& name_view) { return std::string(name_view.data()); });
 }
 
-std::vector<CinnTensor> GetCinnTensorsFromCompiledScope(
-    const std::vector<std::string>& cinn_names, const CinnScope& cinn_scope) {
-  std::vector<CinnTensor> result;
-  result.reserve(cinn_names.size());
-  std::transform(cinn_names.begin(), cinn_names.end(),
-                 std::back_inserter(result),
-                 [&cinn_scope](const std::string& var_name) {
-                   PADDLE_ENFORCE_NOT_NULL(
-                       cinn_scope.FindVar(var_name),
-                       platform::errors::NotFound(
-                           "Variable(%s) not found in cinn scope.", var_name));
-                   return cinn_scope.GetTensor(var_name);
-                 });
-  return result;
+bool CinnLaunchContext::IsVariableUsed(const std::string& paddle_name) {
+  return paddle2cinn_varmap_.count(paddle_name) > 0 &&
+         cinn_variable_names_.count(paddle2cinn_varmap_.at(paddle_name)) > 0;
+}
+
+CinnTensor CinnLaunchContext::GetCinnTensor(const std::string& var_name) {
+  PADDLE_ENFORCE_GT(cinn_variable_names_.count(var_name), 0,
+                    platform::errors::NotFound(
+                        "Variable(%s) not found in cinn scope.", var_name));
+  return cinn_scope_->GetTensor(var_name);
+}
+
+std::vector<std::string> CinnLaunchContext::GetInternalVariableNames() {
+  std::unordered_set<std::string> all_parameters(cinn_variable_names_);
+  std::for_each(paddle2cinn_varmap_.begin(), paddle2cinn_varmap_.end(),
+                [&all_parameters, this](const auto& name_pd2cinn) {
+                  if (IsVariableUsed(name_pd2cinn.first)) {
+                    all_parameters.erase(name_pd2cinn.second);
+                  }
+                });
+  return {all_parameters.begin(), all_parameters.end()};
+}
+
+void CinnLaunchContext::MutableTensorData(const std::string& var_name,
+                                          const platform::Place& place,
+                                          LoDTensor* paddle_tensor,
+                                          bool is_internal_var) {
+  auto cinn_name = var_name;
+  if (!is_internal_var) {
+    PADDLE_ENFORCE_EQ(IsVariableUsed(var_name), true,
+                      platform::errors::InvalidArgument(
+                          "Paddle variable(%s) not used by cinn", var_name));
+    cinn_name = paddle2cinn_varmap_.at(var_name);
+  }
+
+  auto cinn_tensor = GetCinnTensor(cinn_name);
+  // TODO(CtfGo): support mutable corresponding c++ type after CINN ready
+  paddle_tensor->mutable_data<float>(
+      framework::make_ddim(cinn_tensor->shape().data()), place);
 }
 
-void CheckTensorEquivalent(const std::string& paddle_name,
-                           const LoDTensor* paddle_tensor,
-                           const CinnTensor& cinn_tensor) {
+void CinnLaunchContext::CheckTensorEquivalent(const std::string& paddle_name,
+                                              const LoDTensor* paddle_tensor,
```

Comment:
Paddle usually make non-null parameter as const reference instead of const pointer. Could you change the `const LoDTensor* paddle_tensor` to `const LoDTensor& paddle_tensor`?

Reply:
Ok, it is a good suggestion, thank you